core/modules/filter/views/views_handler_field_file_uri.inc

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @file
+ * Definition of views_handler_field_file_uri.
+ * Variant for use in Filter module for Image Library.
+ */
+
+/**
+ * Field handler to add rendering file paths as file URLs instead of as internal file URIs.
+ */
+class views_handler_field_file_uri extends views_handler_field_file {
+  function option_definition() {
+    $options = parent::option_definition();
+    $options['file_download_path'] = array('default' => FALSE, 'bool' => TRUE);
+    $options['image_thumbnail'] = array('default' => FALSE, 'bool' => TRUE);
+    return $options;
+  }
+
+  function options_form(&$form, &$form_state) {
+    $form['file_download_path'] = array(
+      '#title' => t('Display download path instead of file storage URI'),
+      '#description' => t('This will provide the full download URL rather than the internal filestream address.'),
+      '#type' => 'checkbox',
+      '#default_value' => !empty($this->options['file_download_path']),
+    );
+
+    $form['image_thumbnail'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Adapt URI for use of thumbnail'),
+      '#description' => t('Modify the URI to reference the thumbnail image'),
```

Comment:
An example would be helpful here. I don't understand what this means.

Reply:
I have modified the UI and the wording which hopefully now makes it clear that the URL is being modified to link to the selected image style files.